src/core.js

```diff
@@ -299,26 +282,12 @@ function _createWorld({apis, consumers, plugins, upstreams, services, certificat
     getServiceId: serviceName => {
       const id = world.getService(serviceName)._info.id;
 
-      invariant(id, `API ${serviceName} doesn't have an Id`);
-
-      return id;
-    },
-    getApi: apiName => {
-      const api = apis.find(api => api.name === apiName);
-
-      invariant(api, `Unable to find api ${apiName}`);
-
-      return api;
-    },
-    getApiId: apiName => {
-      const id = world.getApi(apiName)._info.id;
-
-      invariant(id, `API ${apiName} doesn't have an Id`);
+      invariant(id, `Service ${serviceName} doesn't have an Id`);
 
       return id;
     },
     getGlobalPlugin: (pluginName, pluginConsumerID) => {
-      const plugin = plugins.find(plugin => plugin.api_id === undefined && plugin.name === pluginName && plugin._info.consumer_id == pluginConsumerID);
```

Comment:
Initially I thought that this `plugin.api_id === undefined` check (and the analogous one in `hasGlobalPlugin`) should be replaced with a test to see if it had a service or route id. It turns out this check was meaningless – the `plugins` array being filtered here is already filtered to only include global plugins in kongState.js.